node/silkworm/rpc/backend_server.cpp

```diff
@@ -58,8 +58,93 @@ void NetVersionService::process_rpc(NetVersionUnaryRpc& rpc, const remote::NetVe
     SILK_TRACE << "NetVersionService::process_rpc rsp: " << &response << " chain_id: " << chain_id_ << " sent: " << sent;
 }
 
+void NetPeerCountService::process_rpc(NetPeerCountRpc& rpc, const remote::NetPeerCountRequest* request) {
+    SILK_TRACE << "NetPeerCountService::process_rpc rpc: " << &rpc << " request: " << request;
```

Comment:
Why print reference to rpc ? I mean `<< &rpc` ?

Reply:
Also ... do `rpc` and `rsp` stand for `request` and `response` ? 
And again ... out of curiosity why don't write 
```
SILK_TRACE << __FUNCTION__ << " rpc: " << &rpc << " request: " << request;
```
At least, should you change function naming, don't have to rewrite in the log lines

Reply:
> Why print reference to rpc ? I mean `<< &rpc` ?

I'm printing the memory address of `rpc` for tracing purposes, it is useful to debug RPC lifetime issues (live RPC objects are allocated and maintained on the heap, then deallocated on cleanup)

Reply:
> Also ... do `rpc` and `rsp` stand for `request` and `response` ?

`rpc` stands for `remote procedure call`, but you have a point here: naming is too much close between `rpc` and `rsp` in the trace log, I will amend it.

> And again ... out of curiosity why don't write
> 
> ```
> SILK_TRACE << __FUNCTION__ << " rpc: " << &rpc << " request: " << request;
> ```
> 
> At least, should you change function naming, don't have to rewrite in the log lines

Yes, you're right, definitely better. Maybe we can go one step further and include `__FUNCTION__` into a couple of extended `SILK_TRACE` logging macros to avoid the repetition altogether:

```
#define SILK_TRACE_ENTER SILK_TRACE << __FUNCTION__ << " ENTER"
#define SILK_TRACE_EXIT  SILK_TRACE << __FUNCTION__ << " EXIT"
```
or

```
#define SILK_TRACE_START SILK_TRACE << __FUNCTION__ << " START"
#define SILK_TRACE_END   SILK_TRACE << __FUNCTION__ << " END"
```

Reply:
Please don't ... 🙏 
Don't add another macro 😄

Reply:
> `rpc` stands for `remote procedure call`, but you have a point here: naming is too much close between `rpc` and `rsp` in the trace log, I will amend it.

Yes I knew the acronym for rpc ... seemed to me you were using it as request response placeholders

Reply:
TBH I'm trying the above suggestion and hit a wall on the class name: I'm realizing that including the class name is really useful to distinguish between `process_rpc` calls in different types, yet AFAICT there is no standard way to obtain demangled class names.

Reply:
Can you build this sample and report the outputs ?

```
$ cat test.cpp 
#include <iostream>

int main(int argc, char **argv)
{
    std::cout << __func__ << std::endl
              << __FUNCTION__ << std::endl
              << __PRETTY_FUNCTION__ << std::endl;
}
$ g++ test.cpp 
$ ./a.out 
```

Reply:
> 

I've just tried a slight variation including some classes to reproduce the class name issue:

https://godbolt.org/z/bh9MPrbeM

None of them seems to work well

Reply:
> Hmmm ... I don't know on linux but at least on windows `__FUNCTION__` returns a FQDN of the function like in this example
> 
> ```
>  ERROR [03-05|19:15:51.470 UTC] HashState                     function=silkworm::stagedsync::HashState::hash_from_account_changeset exception=kAborted
> ```

The problem here is probably that `__FUNCTION__` is not standard

Reply:
MSVC 1 - GCC 0

Reply:
the whole header, or how to get the current function name in any compiler.

```
#ifndef BOOST_CURRENT_FUNCTION_HPP_INCLUDED
#define BOOST_CURRENT_FUNCTION_HPP_INCLUDED

// MS compatible compilers support #pragma once

#if defined(_MSC_VER) && (_MSC_VER >= 1020)
# pragma once
#endif

//
//  boost/current_function.hpp - BOOST_CURRENT_FUNCTION
//
//  Copyright (c) 2002 Peter Dimov and Multi Media Ltd.
//
//  Distributed under the Boost Software License, Version 1.0.
//  See accompanying file LICENSE_1_0.txt or copy at
//  http://www.boost.org/LICENSE_1_0.txt
//
//  [http://www.boost.org/libs/assert/current_function.html](https://www.boost.org/doc/libs/1_63_0/libs/assert/current_function.html)
//

namespace boost
{

namespace detail
{

inline void current_function_helper()
{

#if defined( BOOST_DISABLE_CURRENT_FUNCTION )

# define BOOST_CURRENT_FUNCTION "(unknown)"

#elif defined(__GNUC__) || (defined(__MWERKS__) && (__MWERKS__ >= 0x3000)) || (defined(__ICC) && (__ICC >= 600)) || defined(__ghs__)

# define BOOST_CURRENT_FUNCTION __PRETTY_FUNCTION__

#elif defined(__DMC__) && (__DMC__ >= 0x810)

# define BOOST_CURRENT_FUNCTION __PRETTY_FUNCTION__

#elif defined(__FUNCSIG__)

# define BOOST_CURRENT_FUNCTION __FUNCSIG__

#elif (defined(__INTEL_COMPILER) && (__INTEL_COMPILER >= 600)) || (defined(__IBMCPP__) && (__IBMCPP__ >= 500))

# define BOOST_CURRENT_FUNCTION __FUNCTION__

#elif defined(__BORLANDC__) && (__BORLANDC__ >= 0x550)

# define BOOST_CURRENT_FUNCTION __FUNC__

#elif defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901)

# define BOOST_CURRENT_FUNCTION __func__

#elif defined(__cplusplus) && (__cplusplus >= 201103)

# define BOOST_CURRENT_FUNCTION __func__

#else

# define BOOST_CURRENT_FUNCTION "(unknown)"

#endif

}

} // namespace detail

} // namespace boost

```
#endif // #ifndef BOOST_CURRENT_FUNCTION_HPP_INCLUDED

Reply:
@greg7mdp thanks for your suggestion, but for this use case (including class::method in a log trace) `BOOST_CURRENT_FUNCTION` has the same drawback as `__PRETTY_FUNCTION__` (obviously, because it is the same at least for GCC): too much verbose.

I think that sticking to the hard-coded class::method names in few log traces is not so bad, anyway.